packages/pulp/python-pulpcore/python-pulpcore.spec

```diff
@@ -0,0 +1,94 @@
+# Created by pyp2rpm-3.3.3
+%global pypi_name pulpcore
+
+Name:           python-%{pypi_name}
+Version:        3.0.0rc8
+Release:        1%{?dist}
+Summary:        Pulp Django Application and Related Modules
+
+License:        GPLv2+
+URL:            http://www.pulpproject.org
+Source0:        https://files.pythonhosted.org/packages/source/p/%{pypi_name}/%{pypi_name}-%{version}.tar.gz
+BuildArch:      noarch
+
+BuildRequires:  python3-devel
+BuildRequires:  python3-setuptools
+
+%description
+Pulp is a platform for managing repositories of content, such as software
+packages, and pushing that content out to large numbers of consumers.
+
+Using Pulp you can:
+- Locally mirror all or part of a repository
+- Host your own content in a new repository
+- Manage content from multiple sources in one place
+- Promote content through different repos in an organized way
+
+%package -n     python3-%{pypi_name}
+Summary:        %{summary}
+%{?python_provide:%python_provide python3-%{pypi_name}}
+Requires:       python3-coreapi >= 2.3.3
+Conflicts:      python3-coreapi >= 2.4
+Requires:       python3-djangorestframework-queryfields >= 1.0.0
+Conflicts:      python3-djangorestframework-queryfields >= 1.1
+Requires:       python3-drf-nested-routers >= 0.91
+Conflicts:      python3-drf-nested-routers >= 0.92
+Requires:       python3-drf-yasg >= 1.17.0
+Conflicts:      python3-drf-yasg >= 1.18
+Requires:       python3-rq >= 1.1.0
+Conflicts:      python3-rq >= 1.2
+Requires:       python3-aiofiles
+Requires:       python3-aiohttp
+Requires:       python3-backoff
+Requires:       python3-django >= 2.2.3
+Conflicts:      python3-django >= 2.3
+Requires:       python3-django-filter >= 2.2.0
+Conflicts:      python3-django-filter >= 2.3
+Requires:       python3-django-rest-framework >= 3.10.2
+Conflicts:      python3-django-rest-framework >= 3.11
+Requires:       python3-dynaconf < 2.3
+Requires:       python3-dynaconf >= 2.2
```

Comment:
Related to above, only cause this is damn confusing. And I get that this is auto-generated so we shouldn't have to care. But here is a use of Requires/Requires. And in other places, Requires/Conflicts. How am I supposed to know when to use what?

Reply:
agreed, the mixed usage is confusing at best.   This appears to come down to handling of the pep440  dependencies in setup.py 

https://github.com/pulp/pulpcore/blob/master/setup.py#L13
vs 
https://github.com/pulp/pulpcore/blob/master/setup.py#L23

in other words, it's doing what pulpcore's setup.py told it to do

Reply:
Yes, `~=` gets translated to `Requires/Conflicts`, and an explicit range gets translated to two `Requires`.